metriq_gym/run.py

```diff
@@ -89,11 +95,60 @@ def poll_job(args: argparse.Namespace, job_manager: JobManager) -> None:
     ]
     if all(task.status() == JobStatus.COMPLETED for task in quantum_job):
         result_data: list[ResultData] = [task.result().data for task in quantum_job]
-        print(handler.poll_handler(job_data, result_data))
+        results: BenchmarkResult = handler.poll_handler(job_data, result_data)
+        print(results)
+        if is_upload:
```

Comment:
Another workflow we could support is: - caching results in the jsonl file as a result of a poll action, and then - upload action for results that have already been fetched

Reply:
I am well-familiar with the concept of normalization in dbs, but I don't understand how it applies to my comment.

Reply:
It's a matter having a cache that can fall out-of-sync with the primary source of truth. I'm saying, having a single source of truth for data is good practice in general, not just in SQL databases.

Reply:
Agreed this is out of the scope of this PR anyway. We can have a workflow that does polling for every upload action, and introduced "caching" later on. This is without any overhead of throwaway code.